objects/src/accounts/mod.rs

```diff
@@ -146,12 +147,23 @@ impl ToAdviceInputs for Account {
     fn to_advice_inputs<T: AdviceInputsBuilder>(&self, target: &mut T) {
         // push core items onto the stack
         target.push_onto_stack(&[*self.id, ZERO, ZERO, self.nonce]);
-        target.push_onto_stack(self.vault.root().as_elements());
+        target.push_onto_stack(self.vault.get_commitment().as_elements());
         target.push_onto_stack(&self.storage.root());
         target.push_onto_stack(self.code.root().as_elements());
 
         // extend the merkle store with the storage items
         target.add_merkle_nodes(self.storage.slots().inner_nodes());
-        target.add_merkle_nodes(self.storage.store().inner_nodes())
+        target.add_merkle_nodes(self.storage.store().inner_nodes());
+
+        // extend the merkle store with account vault data
+        target.add_merkle_nodes(self.vault.asset_tree().inner_nodes());
+
+        // populate advice map with tiered merkle tree leaf nodes
+        self.vault.asset_tree().upper_leaves().for_each(|(node, key, value)| {
+            // TODO: temporary hack - assume the node is interted at depth 16 and compute remaining key accordingly
+            let mut key = Word::from(key);
+            key[3] = ((key[3].as_int() << 16) >> 16).into();
+            target.insert_into_map(*node, key.into_iter().chain(value).collect());
+        })
```

Comment:
Should we implement `ToAdviceInputs` for `AccountVault`? This way, the logic of how we put data into the advice provider from the vault is encapsulated there.

Reply:
Yes, good call.